src/python/pants/backend/python/typecheck/mypy/rules_integration_test.py

```diff
@@ -196,6 +196,96 @@ def test_skip(rule_runner: RuleRunner) -> None:
     assert not result
 
 
+def test_thirdparty_dependency(rule_runner: RuleRunner) -> None:
+    rule_runner.add_to_build_file(
+        "",
+        dedent(
+            """\
+            python_requirement_library(
+                name="more-itertools",
+                requirements=["more-itertools==8.4.0"],
+            )
+            """
+        ),
+    )
+    source_file = FileContent(
+        f"{PACKAGE}/itertools.py",
+        dedent(
+            """\
+            from more_itertools import flatten
+
+            assert flatten(42) == [4, 2]
+            """
+        ).encode(),
+    )
+    target = make_target(rule_runner, [source_file])
+    result = run_mypy(rule_runner, [target])
+    assert len(result) == 1
+    assert result[0].exit_code == 1
+    assert f"{PACKAGE}/itertools.py:3" in result[0].stdout
+
+
+def test_thirdparty_plugin(rule_runner: RuleRunner) -> None:
+    rule_runner.add_to_build_file(
+        "",
+        dedent(
+            """\
+            python_requirement_library(
+                name='django',
+                requirements=['Django==2.2.5'],
+            )
+            """
+        ),
+    )
+    settings_file = FileContent(
+        f"{PACKAGE}/settings.py",
+        dedent(
+            """\
+            from django.urls import URLPattern
+
+            DEBUG = True
+            DEFAULT_FROM_EMAIL = "webmaster@example.com"
+            SECRET_KEY = "not so secret"
+            MY_SETTING = URLPattern(pattern="foo", callback=lambda: None)
+            """
+        ).encode(),
+    )
+    app_file = FileContent(
+        f"{PACKAGE}/app.py",
+        dedent(
+            """\
+            from django.utils import text
+
+            assert "forty-two" == text.slugify("forty two")
+            assert "42" == text.slugify(42)
+            """
+        ).encode(),
+    )
+    target = make_target(rule_runner, [settings_file, app_file])
+    config_content = dedent(
+        """\
+        [mypy]
+        plugins =
+            mypy_django_plugin.main
+
+        [mypy.plugins.django-stubs]
+        django_settings_module = project.settings
+        """
+    )
+    result = run_mypy(
+        rule_runner,
+        [target],
+        config=config_content,
+        additional_args=[
+            "--mypy-extra-requirements=django-stubs==1.5.0",
```

Comment:
Note that we install this like we install most plugins - permanently as part of the tool PEX, rather than as a dependency of some code.